src/components/layouts/AdminLayout.tsx

```diff
@@ -0,0 +1,83 @@
+import {
+  Drawer,
+  List,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+} from "@mui/material";
+import InboxIcon from "@mui/icons-material/MoveToInbox";
+import MailIcon from "@mui/icons-material/Mail";
+import { useRouter } from "next/router";
+import { type Route } from "nextjs-routes";
+
+const DRAWER_WIDTH = 240;
+
+type MenuItem = {
+  text: string;
+  icon: JSX.Element;
+  path: "/" | Route;
+};
+
+export default function AdminLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const router = useRouter();
+
+  const menuItems: MenuItem[] = [
+    {
+      text: "Collections",
+      icon: <InboxIcon />,
+      path: "/",
+    },
+    {
+      text: "Forms",
+      icon: <MailIcon />,
+      path: "/",
+    },
+  ];
+
+  const handleNavigation = async (item: MenuItem) => {
+    const path = item.path;
+    await router.push(path);
```

Comment:
Next router seems to return promises for some of these router methods but in all of their examples they don't await. I think this is fine here, although we might want to look at useTransition hook when this is hooked up to actual pages.